Sources/MapboxDirections/MapMatching/Tracepoint.swift

```diff
@@ -1,40 +1,58 @@
 import Foundation
 import CoreLocation
 
-/**
- A `Tracepoint` represents a location matched to the road network.
- */
-public class Tracepoint: Waypoint {
+public extension Match {
     /**
-     Number of probable alternative matchings for this tracepoint. A value of zero indicates that this point was matched unambiguously.
+     A tracepoint represents a location matched to the road network.
      */
-    public let countOfAlternatives: Int
-    
+    struct Tracepoint: Matchpoint, Equatable {
+        /**
+         The geographic coordinate of the waypoint, snapped to the road network.
+         */
+        public var coordinate: CLLocationCoordinate2D
+        
+        /**
+         The straight-line distance from this waypoint to the corresponding waypoint in the `RouteOptions` or `MatchOptions` object.
+         
+         The requested waypoint is snapped to the road network. This property contains the straight-line distance from the original requested waypoint’s `DirectionsOptions.Waypoint.coordinate` property to the `coordinate` property.
+         */
+        public var correction: CLLocationDistance // <-- not in a doc (just for conformance to `Matchpoint`?)
+        
+        /**
+         Number of probable alternative matchings for this tracepoint. A value of zero indicates that this point was matched unambiguously.
+         */
+        public var countOfAlternatives: Int
+        
+        /**
+         The name of the road or path the coordinate snapped to.
+         */
+        public var name: String?
+        
+        /***
+         The index of the match object in matchings that the sub-trace was matched to.
+         */
+        public var matchingIndex: Int = 0 // TODO: not sure about default value
+        
+        /***
+         The index of the waypoint inside the matched route.
+         */
+        public var waypointIndex: Int = 0 // TODO: not sure about default value
```

Comment:
There were not originally in the doc, but looks like it is required to have.

Reply:
These indices are parallel state that can get out of sync with the array of matches or waypoints. This starts to get into #386 territory. I think the filtering suggested in that issue could make these properties unnecessary.